src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj

```diff
@@ -7,6 +7,7 @@
     <OutputType>Library</OutputType>
     <PackageId>Microsoft.NET.Sdk</PackageId>
     <TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
+    <IsShipping>false</IsShipping>
```

Comment:
This package does not ship?

Reply:
Not to nuget.org. The email you sent me mentioned this should be false unless the package was going to nuget.org.

Reply:
What does IsShipping control? We need to have symbols archived and all the rest. We just don't push a package to nuget.org because the ship vehicle is the larger .NET Core SDK.

Reply:
This flag is passed to PushToBlobFeed task that uploads all assets to blob storage. 

https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L70

By shipping to NuGet.org I mean to an official channel (as opposed to unofficial channel like myget). I'll reword the guidelines to be more clear. 

@mmitche to comment on what exactly PushToBlobFeed does with the "NonShipping" flag in artifact manifest.

Reply:
Another thing that's controlled by the flag is localization. We only run XLiffTasks for shipping components.

Reply:
It doesn't do anything with it other than write it into the manifest. In prodcon v1, this then gets interpreted by various tools at release time to determine what needs to be pushed to nuget.org or not.

Reply:
Yes, please. I don't want to be seeing IsShipping=false in the csproj of our product code.

Reply:
In other worlds, the project would declare IsShipping=true since we do ship the bits to an official channel. The fact that the package ships as a part of an installer and not standalone is a publishing detail.

Reply:
Maybe we should just rename the property then?  Really this is saying whether it's an independent shipping unit or not.  For example, nothing in CoreCLR 'ships' in the sense that it independently goes on blob storage or nuget.org, but they do ship in that they end up in other components.  Do not publish is confusing, since you do have to publish for dependency uptake reaosns.

Maybe...ShipsExternally=true/false?

Reply:
@mmitche 
> Really this is saying whether it's an independent shipping unit or not

What do you mean by "this"? The value in the manifest?

Reply:
> Do not publish is confusing, since you do have to publish for dependency uptake reaosns.

Wherever you specify the channel to be nuget.org for pushing various components to, we would specify "no channel" for Microsoft.NET.Sdk.

Reply:
We do not specify what channel a particular component should end up in on project level. Therefore the fact that the component should not be published to any official channel should also not be specified on project level.

In other words, the result of the build (the bits) does not depend on where the component is being published. On the other hand the result of the build depends on whether we consider the component a shipping component or not (e.g. we do localization only for shipping components).

Reply:
> @mmitche
> 
> > Really this is saying whether it's an independent shipping unit or not
> 
> What do you mean by "this"? The value in the manifest?

Yes, the value in the manifest says that.

In prodcon v1 we were specifying where a component ends up on a project level.

In reality, the endpoint isn't actually what we're indicating here. If we suddenly decided to augment nuget.org with Azure DevOps feeds, then we'd have to go and update everywhere to specify the new endpoint.  For a specific output, we really just want the project to indicate in general where the output goes:
- Official package feed
- A storage account layout

It's up to the release process to interpret that information.

Reply:
> It's up to the release process to interpret that information.

How is the release process defined? Could it decide not to publish certain packages?

Reply:
Today it's defined in the tic toc doc.  In general, it would be:
1. Gather list of proposed packages to publish based on the output manifest.
2. Confirm list (manually), looking for:
  - Cases where we missed a package that we know changed.  This might show up in corefx if we forgot to update the list of packages that needed to be published.
  - Cases where a package is being published that shouldn't be.  This is rarer.

The shipping vs. non-shipping gives a first cut.  Take a look at this latest manifest. Of the packages we create, 200+ do not go to nuget.org.

https://github.com/dotnet/versions/blob/master/build-info/dotnet/product/cli/release/2.1/build.xml

Reply:
Do we have a way to specify for some packages that they need to be published somewhere else than to the dotnet-core feed? Or do we expect repos that want to publish somewhere else as well to do that independently? 

What about publishing to Visual Studio? Do we expect to have a different process?

@jaredpar

Reply:
I think we should probably identify those in some way, though not by the exact feed.  To support internal and external builds on equal standing, you need to move the final publish process out of the repo, so that the repo doesn't have to worry about whether a build is public or private, or is stable or not, etc.  That should ideally be handled post-build.  But we could identify feeds in a generic sense like:
- dotnet-core (could correspond to internal or external)
- roslyn

Reply:
Currently we have this in Roslyn: https://github.com/dotnet/roslyn/blob/master/build/config/PublishData.json

It specifies where each kind of artifact should be published. I though we might have something similar on ProdCon v2 level. That's where I was thinking we could also have some exclusions - like  "all shipping packages go here, except for these that are not published as standalone packages"